cmd/algorand-indexer/reindexer.go

```diff
@@ -0,0 +1,57 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/algorand/go-algorand-sdk/types"
+	"github.com/spf13/cobra"
+
+	"github.com/algorand/indexer/config"
+	"github.com/algorand/indexer/idb"
+	"github.com/algorand/indexer/importer"
+)
+
+var (
+	reindexAccount string
+)
+
+var reindexCmd = &cobra.Command{
+	Use:   "reindex",
+	Short: "reindex a single account",
+	Run: func(cmd *cobra.Command, args []string) {
+		var err error
+
+		config.BindFlags(cmd)
+		err = configureLogger()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to configure logger: %v", err)
+			os.Exit(1)
+		}
+
+		opts := idb.IndexerDbOptions{}
+		db := globalIndexerDb(&opts)
+
+		addr, err := types.DecodeAddress(reindexAccount)
+		maybeFail(err, "unable to decode address: %s", reindexAccount)
+
+		maxRounds, err := db.GetMaxRound()
+		maybeFail(err, "unable to fetch max rounds")
+
+		limit := int(maxRounds)
+		filter := idb.UpdateFilter{
+			Address: &addr,
+			StartRound: 0,
+			RoundLimit: &limit,
+		}
+		err = db.DeleteAccount(addr)
```

Comment:
This looks like it should only be run on a database with no daemon running.
Hm, I'd previously balked at trying to put app locking into the database to prevent multiple writers but now I think about it and maybe there's a reasonable way to do it to at least provide a warning.
Add a field to the ImportState which is a timestamp of when it is written.
Add another row to metastate that is an app lock claim, probably it would have start-time and process-id. On graceful exit (^C or kill other than -9) that would be cleared.
On startup check that row, if not present, set one. If present, check ImportState last written time, if that's older than about a minute assume the prior task died, otherwise log error ("looks like there's another writer, exiting") and exit.
This whole interlock system is probably out of scope for this change and could be a separate issue/PR

Reply:
Yes, this command is really dangerous. I made sure to set the "Hidden" flag so it doesn't show up in the help and should probably remove it altogether (or move it to a different root command) before merging any of this work in.

Some sort of locking scheme would be great.